src/composables/transactionData.ts

```diff
@@ -182,11 +189,65 @@ export function useTransactionData({
   );
 
   /**
-   * Amount and fee calculated based on the direction.
+   * TODO most of the following calculations should be done right after fetching the transaction
    */
-  const amountTotal = computed(
-    (): number => (transaction.value) ? getTxAmountTotal(transaction.value, direction.value) : 0,
-  );
+  function getAmountFromAeTx(isReceived: boolean = false): number {
+    if (isDexAllowance.value) {
+      return toShiftedBigNumber(fee.value, -AE_COIN_PRECISION).toNumber();
+    }
+
+    const contractCallData = transaction.value?.tx
+      && categorizeContractCallTxObject(transaction.value);
+
+    const tokenData = contractCallData
+      && getProtocolAvailableTokens(PROTOCOLS.aeternity)[contractCallData.assetContractId!];
+
+    if (contractCallData && tokenData) {
+      return +toShiftedBigNumber(
+        contractCallData.amount || 0,
+        -(tokenData.decimals || AE_COIN_PRECISION), // TODO possibility of wrong precision
+      );
+    }
+
+    const claimTipAmount = (outerTx.value?.function === 'claim') ? outerTx.value?.log?.[0]?.topics[2] : null;
+    const rawAmount = innerTx.value.amount || innerTx.value?.nameFee || claimTipAmount || 0;
+    const amount: BigNumberPublic = (typeof rawAmount === 'object')
+      ? rawAmount
+      : new BigNumber(Number(rawAmount));
+    const gasCost = (outerTx.value?.gasPrice && outerTx.value?.gasUsed)
+      ? new BigNumber(outerTx.value.gasPrice).multipliedBy(outerTx.value.gasUsed)
+      : 0;
+
+    return +aettosToAe(
+      amount
+        .plus(isReceived ? 0 : outerTx.value?.fee || 0)
+        .plus(isReceived ? 0 : innerTx.value?.fee || 0)
+        .plus(isReceived ? 0 : gasCost),
+    );
+  }
+
+  function getAmountFromCommonTx(isReceived: boolean = false): number {
+    return new BigNumber(outerTx.value?.amount || 0)
+      .plus(isReceived ? 0 : fee.value)
+      .toNumber();
+  }
+
+  /**
+   * Amount without fee
+   */
+  const amount = computed((): number => (protocol.value === PROTOCOLS.aeternity)
+    ? getAmountFromAeTx()
+    : getAmountFromCommonTx());
+
+  /**
+   * Amount with the fee added when receiving the assets
+   */
+  const amountTotal = computed((): number => {
+    const isReceived = direction.value === TX_DIRECTION.received;
```

Comment:
`isReceived` could be a computed property and be used throughout the composable. It could also be exported from the composable and used in other places like `TransactionDetails`